apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -341,28 +374,82 @@ impl ApplyTo for PathSelection {
     }
 }
 
+impl ApplyTo for JSLiteral {
```

Comment:
Tying ourselves implicitly to JavaScript feels weird. Do we need all the `JS` namespaces, or can we just call these `Literal`, `Object`, etc.—saying "these are the literals for _our_ language"?

Reply:
I think an appropriate generic name for this type would be `LiteralExpression`, since these values are not immediately JSON, but can be turned into concrete JSON once you provide the input variables.

Reply:
Went with `LitExpr` for the main `JSLiteral` type, as well as `Lit*` prefixes for the other types (e.g. `LitObject`): https://github.com/apollographql/router/pull/5762/commits/c45b06832c056ab19e791e24b2dc99a4272932f5